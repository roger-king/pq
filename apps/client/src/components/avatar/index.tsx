import { AvatarIcon } from './avatar';

export default AvatarIcon;
