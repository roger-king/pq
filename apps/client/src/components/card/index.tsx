export * from './question';
export * from './game';
