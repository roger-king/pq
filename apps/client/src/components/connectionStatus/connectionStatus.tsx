import React from 'react';
import { Box, Text } from 'grommet';

export interface ConnectionStatusProps {
  connected: boolean;
}

export const ConnectionStatus: React.FC<ConnectionStatusProps> = ({ connected }) => {
  return (
    <Box direction="row" gap="small" align="center" justify="center">
      <Box background={connected ? 'green' : 'red'} style={{ borderRadius: '50px' }} width="15px" height="15px" />
      <Text>{connected ? 'Connected' : 'Not Connected'}</Text>
    </Box>
  );
};
