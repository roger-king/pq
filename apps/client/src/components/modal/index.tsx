import React, { PropsWithChildren } from 'react';
import { Box, Heading, Layer } from 'grommet';

export interface BaseModalProps {
  title: string;
  onClose(): void;
}

const Modal: React.FC<PropsWithChildren<BaseModalProps>> = (props: PropsWithChildren<BaseModalProps>) => {
  const { children, title } = props;

  return (
    <Layer modal>
      <Box fill align="center" justify="center" pad="medium" background="brand">
        <Box direction="row" align="center" justify="between" fill>
          <Heading level="3">{title}</Heading>
        </Box>
        {children}
      </Box>
    </Layer>
  );
};

export default Modal;
