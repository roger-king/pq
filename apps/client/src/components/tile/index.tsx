import { Tile } from './tile.component';

export default Tile;
