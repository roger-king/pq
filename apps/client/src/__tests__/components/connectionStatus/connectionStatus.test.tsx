import React from 'react';
import renderer from 'react-test-renderer';

import { ConnectionStatus } from '../../../components/connectionStatus';

describe('<ConnectionStatus />', () => {
  test('should display connected ConnectionStatus', async () => {
    const tree = renderer.create(<ConnectionStatus connected />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should display disconnected ConnectionStatus', async () => {
    const tree = renderer.create(<ConnectionStatus connected={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should display ConnectionStatus with Children', async () => {
    const tree = renderer
      .create(
        <ConnectionStatus connected={false}>
          <div>Hello</div>
        </ConnectionStatus>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
