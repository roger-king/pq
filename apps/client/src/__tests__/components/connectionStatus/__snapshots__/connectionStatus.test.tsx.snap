// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectionStatus /> should display ConnectionStatus with Children 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 iQEJEP"
>
  <div
    className="StyledBox-sc-13pk1d4-0 cOvDVi"
    style={
      Object {
        "borderRadius": "50px",
      }
    }
  />
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
  />
  <span
    className="StyledText-sc-1sadyjn-0 DkZvS"
  >
    Not Connected
  </span>
</div>
`;

exports[`<ConnectionStatus /> should display connected ConnectionStatus 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 iQEJEP"
>
  <div
    className="StyledBox-sc-13pk1d4-0 fcIOFG"
    style={
      Object {
        "borderRadius": "50px",
      }
    }
  />
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
  />
  <span
    className="StyledText-sc-1sadyjn-0 DkZvS"
  >
    Connected
  </span>
</div>
`;

exports[`<ConnectionStatus /> should display disconnected ConnectionStatus 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 iQEJEP"
>
  <div
    className="StyledBox-sc-13pk1d4-0 cOvDVi"
    style={
      Object {
        "borderRadius": "50px",
      }
    }
  />
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
  />
  <span
    className="StyledText-sc-1sadyjn-0 DkZvS"
  >
    Not Connected
  </span>
</div>
`;
